core/src/main/java/org/springframework/security/core/userdetails/ReactiveUserDetailsChecker.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.core.userdetails;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * This is a reactive counterpart of {@link UserDetailsChecker}. To use existing an checker,
+ * refer to {@link org.springframework.security.authentication.ReactiveDelegatingUserDetailsChecker}
+ *
+ * @author Alexey Nesterov
+ * @since 5.2
+ */
+public interface ReactiveUserDetailsChecker {
```

Comment:
I don't know that we need a reactive equivalent since the UserDetails has already been looked up and thus everything is non blocking. This does presume that nothing additional is needed to be looked up, but I think this is a pretty safe assumption given the API is for checking a UserDetails. If additional details are necessary it can operate on a custom implementation of UserDetails which contains the additional properties that need to be checked.

Reply:
It is gone now. I wasn't really sure about this one, my confusion was related to the fact that current checker throws exceptions, rather than returning a value. However more I look into that, more I understand that shifting from exception is much bigger task.